src/TextDrawingComponent.cpp

```diff
@@ -0,0 +1,316 @@
+				break;
+
+				case isobus::TextualVTObject::VerticalJustification::PositionBottom:
+				{
+					retVal = Justification::bottomRight;
+				}
+				break;
+
+				case isobus::TextualVTObject::VerticalJustification::Reserved:
+				default:
+					break;
+			}
+		}
+		break;
+
+		default:
+		{
+		}
+		break;
+	}
+	return retVal;
+}
+
+std::uint8_t TextDrawingComponent::prepare_text_painting(Graphics &g,
+                                                         std::shared_ptr<isobus::FontAttributes> font,
+                                                         char referenceCharForWidthCalc)
+{
+	Font juceFont = Font(Font::getDefaultMonospacedFontName(), 0, 0);
+	std::uint8_t fontHeight;
+	auto fontType = font->get_type();
+	auto colour = parentWorkingSet->get_colour(font->get_colour());
+	int fontStyleFlags = Font::FontStyleFlags::plain;
+
+	if (font->get_style(isobus::FontAttributes::FontStyleBits::Bold))
+	{
+		fontStyleFlags |= Font::FontStyleFlags::bold;
+	}
+
+	if (font->get_style(isobus::FontAttributes::FontStyleBits::Italic))
+	{
+		fontStyleFlags |= Font::FontStyleFlags::italic;
+	}
+
+	if (font->get_style(isobus::FontAttributes::FontStyleBits::Underlined))
+	{
+		fontStyleFlags |= Font::FontStyleFlags::underlined;
+	}
+
+	juceFont.setStyleFlags(fontStyleFlags);
+	juceFont.setHeight(font->get_font_height_pixels());
+
+	auto fontWidth = juceFont.getStringWidthFloat(juce::String::fromUTF8(&referenceCharForWidthCalc, 1));
+	fontHeight = font->get_font_width_pixels();
+	if (fontHeight == 0)
+	{
+		fontHeight = 8;
+	}
+	juceFont.setHorizontalScale(static_cast<float>(font->get_font_width_pixels()) / fontWidth);
+
+	g.setColour(Colour::fromFloatRGBA(colour.r, colour.g, colour.b, 1.0f));
+	g.setFont(juceFont);
+
+	return fontHeight;
+}
+
+void TextDrawingComponent::drawStrikeThrough(Graphics &g, int w, int h, const String &str, isobus::TextualVTObject::HorizontalJustification justification)
+{
+	auto font = g.getCurrentFont();
+	auto textWidth = font.getStringWidth(str);
+	auto lineThickness = h * 0.05; // set the thickness to 5% of the total text height
+	if (lineThickness < 1.0)
+	{
+		lineThickness = 1.0;
+	}
+
+	auto y = h / 2;
+
+	switch (justification)
+	{
+		case isobus::TextualVTObject::HorizontalJustification::PositionLeft:
+			g.drawLine(0, y, textWidth, y, lineThickness);
+			break;
+		case isobus::TextualVTObject::HorizontalJustification::PositionMiddle:
+			g.drawLine(w / 2 - textWidth / 2, y, w / 2 + textWidth / 2, y, lineThickness);
+			break;
+		case isobus::TextualVTObject::HorizontalJustification::PositionRight:
+			g.drawLine(w - textWidth, y, w, y, lineThickness);
+			break;
+		default:
+			break;
+	}
+}
+
+void TextDrawingComponent::paintText(Graphics &g, const std::string &text, bool enabled)
+{
+	std::string value = text;
+	std::uint8_t fontHeight = 8;
+	bool strikeThrough = false;
+	auto fontType = isobus::FontAttributes::FontType::ISO8859_1;
+	auto sourceString = static_cast<isobus::StringVTObject *>(sourceObject);
```

Comment:
Probably need a type check here or an assert if the supplied base object is the wrong object flavor